qdev_wrappers/dataset/doNd.py

```diff
@@ -0,0 +1,278 @@
+          will be called at each step. The function should take no arguments.
+          The parameters and functions are called in the order they are
+          supplied.
+        enter_actions: A list of functions taking no arguments that will be
+            called before the measurements start
+        exit_actions: A list of functions taking no arguments that will be
+            called after the measurements ends
+        do_plot: should png and pdf versions of the images be saved after the
+            run.
+
+    Returns:
+        The run_id of the DataSet created
+    """
+    meas = Measurement()
+    meas.register_parameter(
+        param_set)  # register the first independent parameter
+    output = []
+    param_set.post_delay = delay
+    interrupted = False
+
+    for action in enter_actions:
+        # this omits the posibility of passing
+        # argument to enter and exit actions.
+        # Do we want that?
+        meas.add_before_run(action, ())
+    for action in exit_actions:
+        meas.add_after_run(action, ())
+
+    # do1D enforces a simple relationship between measured parameters
+    # and set parameters. For anything more complicated this should be
+    # reimplemented from scratch
+    for parameter in param_meas:
+        if isinstance(parameter, _BaseParameter):
+            meas.register_parameter(parameter, setpoints=(param_set,))
+            output.append([parameter, None])
+
+    try:
+        with meas.run() as datasaver:
+
+            for set_point in np.linspace(start, stop, num_points):
+                param_set.set(set_point)
+                output = []
+                for i, parameter in enumerate(param_meas):
+                    if isinstance(parameter, _BaseParameter):
+                        output.append((parameter, parameter.get()))
+                    elif callable(parameter):
+                        parameter()
+                datasaver.add_result((param_set, set_point),
+                                      *output)
+    except KeyboardInterrupt:
+        interrupted = True
+
+    dataid = datasaver.run_id  # convenient to have for plotting
+
+    if do_plot is True:
+        ax, cbs = _save_image(datasaver)
+    else:
+        ax = None,
+        cbs = None
+
+    if interrupted:
+        raise KeyboardInterrupt
+    return dataid, ax, cbs
+
+
+def do2d(param_set1: _BaseParameter, start1: number, stop1: number,
+         num_points1: int, delay1: number,
+         param_set2: _BaseParameter, start2: number, stop2: number,
+         num_points2: int, delay2: number,
+         *param_meas: _BaseParameter,
+         enter_actions: Sequence[Callable[[], None]] = (),
+         exit_actions: Sequence[Callable[[], None]] = (),
+         before_inner_actions: Sequence[Callable[[], None]] = (),
+         after_inner_actions: Sequence[Callable[[], None]] = (),
+         do_plot: bool=True) -> AxesTupleListWithRunId:
```

Comment:
After all you decided against implementing the `set_before_sweep` feature? I am not sure how much it is actually used...